providers/vercel/api.go

```diff
@@ -0,0 +1,170 @@
+
+// ListDNSRecords retrieves all DNS records for a domain, handling pagination.
+func (c *vercelProvider) ListDNSRecords(ctx context.Context, domain string) ([]DNSRecord, error) {
+	var allRecords []DNSRecord
+	var nextTimestamp int64
+
+	for {
+		url := fmt.Sprintf("https://api.vercel.com/v4/domains/%s/records?limit=%d", domain, vercelAPIPaginationLimit)
+		if c.teamID != "" {
+			url += fmt.Sprintf("&teamId=%s", c.teamID)
+		}
+		if nextTimestamp != 0 {
+			url += fmt.Sprintf("&until=%d", nextTimestamp)
+		}
+
+		var result listResponse
+		err := c.doRequest(clientRequest{
+			ctx:    ctx,
+			method: "GET",
+			url:    url,
+		}, &result, c.listLimiter)
+
+		if err != nil {
+			return nil, fmt.Errorf("failed to list DNS records: %w", err)
+		}
+
+		for _, r := range result.Records {
+			// The official SDK expects 'recordType' but the API returns 'type'.
+			// We explicitly map it here to fix the discrepancy.
+			r.RecordType = r.Type
+			// Ensure Domain field is set (it might not be in the record object itself)
+			if r.Domain == "" {
+				r.Domain = domain
+			}
+			if r.TeamID == "" {
+				r.TeamID = c.teamID
+			}
+
+			allRecords = append(allRecords, r)
+		}
+
+		if result.Pagination.Next == nil {
+			break
+		}
+		nextTimestamp = *result.Pagination.Next
+	}
+
+	return allRecords, nil
+}
+
+// httpsRecord structure for Vercel API
+type httpsRecord struct {
+	Priority int64  `json:"priority"`
+	Target   string `json:"target"`
+	Params   string `json:"params,omitempty"`
+}
+
+// createDNSRecordRequest embeds the official SDK request but adds HTTPS support
+type createDNSRecordRequest struct {
+	vercelClient.CreateDNSRecordRequest
+	HTTPS *httpsRecord `json:"https,omitempty"`
+}
+
+// CreateDNSRecord creates a DNS record.
+func (c *vercelProvider) CreateDNSRecord(ctx context.Context, req createDNSRecordRequest) (*vercelClient.DNSRecord, error) {
+	url := fmt.Sprintf("https://api.vercel.com/v4/domains/%s/records", req.Domain)
+	if c.teamID != "" {
+		url += fmt.Sprintf("?teamId=%s", c.teamID)
+	}
+
+	var response struct {
+		RecordID string `json:"uid"`
+	}
+
+	payloadJSON, err := json.Marshal(req)
+	if err != nil {
+		return nil, err
+	}
+
+	err = c.doRequest(clientRequest{
+		ctx:    ctx,
+		method: "POST",
+		url:    url,
+		body:   string(payloadJSON),
+	}, &response, c.createLimiter)
+	if err != nil {
+		return nil, err
+	}
+
+	return &vercelClient.DNSRecord{ID: response.RecordID}, nil
+}
+
+// updateDNSRecordRequest embeds the official SDK request but adds HTTPS support
+type updateDNSRecordRequest struct {
+	vercelClient.UpdateDNSRecordRequest
+	HTTPS *httpsRecord `json:"https,omitempty"`
+}
+
+// UpdateDNSRecord updates a DNS record.
+func (c *vercelProvider) UpdateDNSRecord(ctx context.Context, recordID string, req updateDNSRecordRequest) (*vercelClient.DNSRecord, error) {
+	url := fmt.Sprintf("https://api.vercel.com/v4/domains/records/%s", recordID)
+	if c.teamID != "" {
+		url += fmt.Sprintf("?teamId=%s", c.teamID)
```

Comment:
```suggestion
		url += "?teamId=" + c.teamID
```

Reply:
Done in [`bdc5cee` (#3542)](https://github.com/StackExchange/dnscontrol/pull/3542/commits/bdc5cee9a7964c6723bd14d35950219f1c409744).